### PR TITLE
4195 mobile lighthouse ccg letters

### PR DESCRIPTION
**Pull request by aherzberg**
## Description of change

A redo of what I had to previously revert due to some bad settings. won't merge this until the corresponding Devops PRs are ready to merge as well. will link those once they are up. 

Only changes that were made from what was reverted was a separation of the immunization and letters tokens. 

Previous revert: https://github.com/department-of-veterans-affairs/vets-api/pull/11281

Devops PR: https://github.com/department-of-veterans-affairs/devops/pull/12325

### CODE REVIEW

File: modules/mobile/spec/request/letters_lighthouse_request_spec.rb
Comment (kpethtel) [inline modules/mobile/spec/request/letters_lighthouse_request_spec.rb:79]: I'm a little confused by this. Is this intended to overwrite the stub in the before block?

**Review by kpethtel — DISMISSED** [review]
Few small questions, but nothing blocking. My guess is that flipper automatically re-enables, but I'm not totally sure. I'd be surprised if we're not disabling it without re-enabling it elsewhere, so it's probably fine and we're just being inconsistent in how we use it.

File: modules/mobile/app/controllers/mobile/v0/letters_controller.rb
Comment (aherzberg) [reply modules/mobile/app/controllers/mobile/v0/letters_controller.rb:10]: some of the comments from the original PR for this spoke to this. I confirmed with Lighthouse that we only need to verify the user has a ICN which our letters service already does (and all our users should have an icn anyway). before it was up in the air so I kept it just to make sure.

File: modules/mobile/spec/request/letters_lighthouse_request_spec.rb
Comment (aherzberg) [reply modules/mobile/spec/request/letters_lighthouse_request_spec.rb:79]: I was banging by head against this one for a while. the schema validation code, calls a `read` on a `File` and kept failing on me. It was only when I realized that it was returning the rsa key from above instead of the actual schema json file that I realized that the stub needed to be reversed. I tried playing with more elegant ways of doing this (like using `.with`) but couldn't find another solution.

Comment (kpethtel) [reply modules/mobile/spec/request/letters_lighthouse_request_spec.rb:79]: That makes sense. Given the way this spec file is structured, couldn't you just move the stub from the before block above into the error handling context. Isn't that where the stub is actually needed?

File: modules/mobile/spec/request/user_request_spec.rb
Comment (aherzberg) [reply modules/mobile/spec/request/user_request_spec.rb]: When the flag is on and we're using LH for letters, we don't need to validate anything but FE still needs to see it as an authorized service since they won't be making any FE changes for this new upstream service.

File: modules/mobile/spec/request/user_request_spec.rb
Comment (aherzberg) [reply modules/mobile/spec/request/user_request_spec.rb]: oh that good to know. I moved the disable into the individual specs that use it instead.

**Review by kpethtel — APPROVED** [review]

**Comment by holdenhinkle** [comment]
@aherzberg Can we please put this PR and [this PR](https://github.com/department-of-veterans-affairs/devops/pull/12325) in draft mode until the later `devops` PR is ready? They're in my list to review and I keep looking at them only to find the `devops` PR isn't ready. Thanks.

**Comment by aherzberg** [comment]
> @aherzberg Can we please put this PR and [this PR](https://github.com/department-of-veterans-affairs/devops/pull/12325) in draft mode until the later `devops` PR is ready? They're in my list to review and I keep looking at them only to find the `devops` PR isn't ready. Thanks.

Sure. Not seeing a way to put back in draft but I'll close them for now.

**Comment by aherzberg** [comment]
@holdenhinkle since all behavior changes are behind a flipper, I will merge this in before the devops changes are in so I can continue development on the rest of these letter endpoints while waiting on devops feedback.

**Comment by holdenhinkle** [comment]
> > @aherzberg Can we please put this PR and [this PR](https://github.com/department-of-veterans-affairs/devops/pull/12325) in draft mode until the later `devops` PR is ready? They're in my list to review and I keep looking at them only to find the `devops` PR isn't ready. Thanks.
> 
> Sure. Not seeing a way to put back in draft but I'll close them for now.

Here's how:
image

**Comment by aherzberg** [comment]
> > > @aherzberg Can we please put this PR and [this PR](https://github.com/department-of-veterans-affairs/devops/pull/12325) in draft mode until the later `devops` PR is ready? They're in my list to review and I keep looking at them only to find the `devops` PR isn't ready. Thanks.
> > 
> > 
> > Sure. Not seeing a way to put back in draft but I'll close them for now.
> 
> Here's how: image

oh got it. Sorry, I missed that. I'll make sure to do that for next time.

**Review by holdenhinkle — APPROVED** [review]
Approving this, but please respond to @kpethtel's comments before merging. Ping me if you commit changes and need another review.

File: modules/mobile/spec/request/letters_lighthouse_request_spec.rb
Comment (aherzberg) [reply modules/mobile/spec/request/letters_lighthouse_request_spec.rb:79]: sorry, forgot to respond to this one. you're right. it shouldn't be nested so far down but when I try to bring it up, it starts failing again and this time I'm not sure exactly why. there must be other File reads being done with VCR. I'm gonna keep it there for now but look more into it for the future letters PR that I'll be creating today or tomorrow. I'm just surprised we haven't ran into this issue before since we often compare the schemas and use that RSA file read stub

**Pull request merged**
